docs/exercises/01-iteration/03-filter.exercise.ts

```diff
@@ -13,7 +13,7 @@ import { TODO } from '../src/todo';
 import { run } from '../src/example';
 import { collecting } from '../src/collecting';
 
-function filterNullish<T>(iterable: Iterable<T>): Iterable<NonNullable<T>> {
+function filterNullish<T>(iterable: Iterable<T>): Array<NonNullable<T>> {
```

Comment:
Having it return an `Iterable` was intentional as this exercise is trying to nudge people away from always creating iterator sinks–and thus realizing all values–at every function boundary. Was it broken somehow for you?

Reply:
Yeah, it wasn't working for me. With this solution:

<img width="775" alt="Screenshot 2024-01-02 at 11 51 55 AM" src="https://github.com/votingworks/vxsuite/assets/530106/1dcf74a4-cf5a-4c82-a98b-fcc92d4509a8">

I got this test failure:
<img width="777" alt="Screenshot 2024-01-02 at 11 52 01 AM" src="https://github.com/votingworks/vxsuite/assets/530106/736686ce-e93b-44cc-897d-8f94ac2d3de5">

In typing this comment out I ended up figuring out that the actual fix is this:

```
diff --git a/docs/exercises/01-iteration/03-filter.exercise.ts b/docs/exercises/01-iteration/03-filter.exercise.ts
index a766f7185..1263d0b32 100644
--- a/docs/exercises/01-iteration/03-filter.exercise.ts
+++ b/docs/exercises/01-iteration/03-filter.exercise.ts
@@ -34,7 +34,7 @@ function filterNullishReference<T>(
 run({
   makeInput: () => [1, 2, null, 3, undefined, 4],
   referenceImplementation: filterNullishReference,
-  exerciseImplementation: filterNullish,
+  exerciseImplementation: collecting(filterNullish),
   solutionImplementation: collecting(filterNullishSolution),
 });
```

Reply:
It seems to work if you make all of them use `collecting`:

```ts
run({
  makeInput: () => [1, 2, null, 3, undefined, 4],
  referenceImplementation: collecting(filterNullishReference),
  exerciseImplementation: collecting(filterNullish),
  solutionImplementation: collecting(filterNullishSolution),
});
```

Reply:
Nice! Fixed and fixuped